lib/classification_lifecycle.rb

```diff
@@ -1,82 +1,106 @@
 class ClassificationLifecycle
   class ClassificationNotPersisted < StandardError; end
 
-  attr_reader :classification
-
-  def initialize(classification)
-    @classification = classification
-  end
-
-  def queue(action)
+  def self.queue(classification, action)
     unless classification.persisted?
       message = "Background process called before persisting the classification."
       raise ClassificationNotPersisted.new(message)
     end
     ClassificationWorker.perform_async(classification.id, action.to_s)
   end
 
-  def transact!(&block)
+  def self.perform(classification, action)
+    raise "Invalid Post-Classification Action" unless %w(create update).include?(action)
+    ClassificationLifecycle.new(classification, action).execute
+  end
+
+  attr_reader :classification, :action
+
+  def initialize(classification, action)
+    @classification = classification
+    @action = action
+  end
+
+  def execute
+    return if create? && classification.lifecycled_at.present?
+
     Classification.transaction do
       update_classification_data
-      #NOTE: ensure the block is evaluated before updating the seen subjects
-      # as the count worker won't fire if the seens are set, see #should_count_towards_retirement
-      instance_eval(&block) if block_given?
+      update_counters
+      process_project_preference
       create_recent
       update_seen_subjects
     end
+
     publish_data
+    notify_cellect
```

Comment:
can this be above the publish data to get into the active sidekiq slots (hopefully / depends on queue priority) before the publish jobs.